### PR TITLE
Add Roblox app

### PR DESCRIPTION
**Pull request by C0DESTR0NG**
# Roblox for Tidbyt

Displays an online friends or favorite games view for a specified Roblox username.

A pulsating green dot will appear if the Roblox user is online, otherwise it will show gray for offline. Friend avatars also provide an online vs offline dot indicator.

### ONLINE FRIENDS VIEW

Sample Online Friends View

### FAVORITE GAMES VIEW

Sample Favorite Games View

### Customization

This app provides various UI options. The user can customize their experience by choosing to show their online friends vs. favorite games views, selecting an accent text color or enabling dark vs. light modes.  

These customizations can be adjusted using the properties of the app.

### CODE REVIEW

**Comment by C0DESTR0NG** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by matslina — CHANGES_REQUESTED** [review]
Hey, this looks really promising! The layout is very spiffy! =)

There are some issues with caching that I think we need addressed before merging. See the individual code comments for more context. If you find this confusing, feel free to ask questions in the comments here, or hit me up via DM on discord.

File: apps/roblox/roblox.star [hunk 1]
```diff
@@ -0,0 +1,620 @@
...
+        print("Found new user name")
+        clearCaches()
+
+    cache.set("cached_user_name", config.str("username") if config.str("username") != None and config.str("username") != "" else DEFAULT_USER_NAME, ttl_seconds = TTL_SECONDS)
+
+    ### SET USERNAME
+    username = config.str("username") if config.str("username") != None and config.str("username") != "" else DEFAULT_USER_NAME
+
+    ### GET USER ID
+    user_id_cached = cache.get("user_id")
```
[the remark this answers is not in the document]
Comment (C0DESTR0NG) [reply apps/roblox/roblox.star]: Great call out. I removed clearCaches() and updated the caches to use username as a key per your request. Now included in latest commit.

File: apps/roblox/roblox.star [hunk 2]
```diff
@@ -0,0 +1,620 @@
...
+            print("Fetching user favorite game list")
+            getUsersFavoriteGames = "https://games.roblox.com/v2/users/%s/favorite/games?accessFilter=Public&sortOrder=Desc&limit=10" % userRobloxId
+            repGetUsersFavoriteGames = http.get(getUsersFavoriteGames)
+            if repGetUsersFavoriteGames.status_code != 200:
+                print("Fetching user favorite game list failed with status %d" % repGetUsersFavoriteGames.status_code)
+                userFavoriteGames = []
+            else:
+                userFavoriteGames = repGetUsersFavoriteGames.json()["data"]
+
+        cache.set("user_favorite_games_list", json.encode(userFavoriteGames), ttl_seconds = TTL_SECONDS)
```
[the remark this answers is not in the document]
Comment (C0DESTR0NG) [reply apps/roblox/roblox.star]: These cache setters have also been updated in the latest commit 👍🏼

File: apps/roblox/roblox.star [hunk 3]
```diff
@@ -0,0 +1,620 @@
...
+        print("Fetching user avatar url")
+        getProfilePhoto = "https://thumbnails.roblox.com/v1/users/avatar-headshot?userIds=%s&size=60x60&format=Png&isCircular=true" % userRobloxId
+        repGetProfilePhoto = http.get(getProfilePhoto)
+        if repGetProfilePhoto.status_code != 200:
+            print("Fetching user avatar url failed with status %d" % repGetProfilePhoto.status_code)
+            profilePhotoUrl = ""
+        else:
+            profilePhotoUrl = repGetProfilePhoto.json()["data"][0]["imageUrl"]
+
+    profilePhotoImg = http.get(profilePhotoUrl).body() if profilePhotoUrl != "" else ""
```
[the remark this answers is not in the document]
Comment (C0DESTR0NG) [reply apps/roblox/roblox.star]: @matslina Now caching **all** _user_, _friend_ and _game_ image data by key. Seeing much faster load times between app config updates. Thank you!

**Review by matslina — APPROVED** [review]

**Pull request merged**